src/nsresourced/meson.build

```diff
@@ -0,0 +1,48 @@
+# SPDX-License-Identifier: LGPL-2.1-or-later
+
+subdir('bpf/userns_restrict')
+
+systemd_nsresourcework_sources = files(
+        'nsresourcework.c',
+        'userns-restrict.c',
+        'userns-registry.c',
+)
+
+systemd_nsresourced_sources = files(
+        'nsresourced-manager.c',
+        'nsresourced.c',
+        'userns-restrict.c',
+        'userns-registry.c',
+)
+
+userns_restrict_include = include_directories('.')
+
+if conf.get('HAVE_VMLINUX_H') == 1
```

Comment:
Given the BPF LSM is pretty much fundamental for the security model of this component, shouldn't this be disabled completely if bpf lsm support is not built in? And then avoid the specific meson_options.txt knob?

Reply:
we need the bpf stuff only for the *unpriv* userns allocation stuff. But some of the resources the service delegates are also accessible to regular users and priv users, so i figured we should leave it on.

for example, right now nspawn asks for a delegated netif when running unpriv. I think we should use that in vmspawn too eventually, because the semantics are super similar. But in that case there would be no userns involved at all.

Reply:
Ok, are the other cases rejected when bpf support is not built in? I didn't spot it but I might just have missed it, a comment documenting it would be useful I think

Reply:
Yes, we have three checks in place: we pass an env var to the workers that indicate if we should expose the userns stuff, which directly maps to the bpf feature being available. if we pass that check, then on each request we install the bpf hooks again (which is a NOP if already installed, but will fail if bpf is not available), and then we add the userns we allowlist to a bpf map before we allow it through, and that would fail too...

so i think we should be fine, it's very hard to even accidentally make nsresourced do something without the bpf stuff being fully in effect.

I'll add a comment clarifying this situation to the primary knob, i.e. the the function test_userns_api_support() that checks for $NSRESOURCE_API.